File: app/src/main/java/com/adsamcik/signalcollector/file/CacheStore.java
package com.adsamcik.signalcollector.file;

import android.content.Context;
import android.support.annotation.NonNull;

import java.io.File;

public class CacheStore {

	public static File file(@NonNull Context context, @NonNull String fileName) {
		return FileStore.file(context.getCacheDir(), fileName);
	}

	/**
	 * Saves string to file
	 *
	 * @param fileName file name
	 * @param data     string data
	 */
	@SuppressWarnings("SameParameterValue")
	public static boolean saveString(@NonNull Context context, @NonNull String fileName, @NonNull String data, boolean append) {
		return FileStore.saveString(file(context, fileName), data, append);
	}


	/**
	 * Load string file as StringBuilder
	 *
	 * @param fileName file name
	 * @return content of file as StringBuilder
	 */
	public static StringBuilder loadStringAsBuilder(@NonNull Context context, @NonNull String fileName) {
		return FileStore.loadStringAsBuilder(file(context, fileName));
	}

	/**
	 * Converts loadStringAsBuilder to string and handles nulls
	 *
	 * @param fileName file name
	 * @return content of file (empty if file has no content or does not exists)
	 */
	public static String loadString(@NonNull Context context, @NonNull String fileName) {
		return FileStore.loadString(file(context, fileName));
	}

	/**
	 * Checks if file exists
	 *
	 * @param fileName file name
	 * @return existence of file
	 */
	@SuppressWarnings("BooleanMethodIsAlwaysInverted")
	public static boolean exists(@NonNull Context context, @NonNull String fileName) {
		return file(context, fileName).exists();
	}

	public static void clearAll(@NonNull Context context) {
		FileStore.clearFolder(context.getCacheDir());
	}
}

File: app/src/main/java/com/adsamcik/signalcollector/file/CacheStore.kt
package com.adsamcik.signalcollector.file

import android.content.Context

import java.io.File

/**
 * Utility class for storing cache files
 * Uses [FileStore] for work with files
 */
object CacheStore {

    fun file(context: Context, fileName: String): File = FileStore.file(context.cacheDir, fileName)

    /**
     * Saves string to file
     *
     * @param fileName file name
     * @param data     string data
     */
    fun saveString(context: Context, fileName: String, data: String, append: Boolean): Boolean =
            FileStore.saveString(file(context, fileName), data, append)


    /**
     * Converts loadStringAsBuilder to string and handles nulls
     *
     * @param fileName file name
     * @return content of file (empty if file has no content or does not exists)
     */
    fun loadString(context: Context, fileName: String): String? =
            FileStore.loadString(file(context, fileName))

    /**
     * Checks if file exists
     *
     * @param fileName file name
     * @return existence of file
     */
    fun exists(context: Context, fileName: String): Boolean = file(context, fileName).exists()

    fun clearAll(context: Context) {
        FileStore.clearDirectory(context.cacheDir)
    }
}
